### PR TITLE
Add more testing for RootFileReader class

### PR DESCRIPTION
**Pull request by AndreasAlbert**
Only merge after #58

### CODE REVIEW

**Comment by clelange** [comment]
@AndreasAlbert 
Conflicting file: tests/test_rootfilereader.py

**Comment by AndreasAlbert** [comment]
Hi @clelange the merge conflicts have been resolved. It seems CI seems to fail because the OSX build times out, as it does in other PRs. Can we please merge this?

File: tests/test_rootfilereader.py [hunk 1]
```diff
@@ -7,13 +7,44 @@
...
+        # Check with wrong input type
+        with self.assertRaises(ValueError):
+            reader = RootFileReader(5)
+        with self.assertRaises(ValueError):
+            reader = RootFileReader([])
+        with self.assertRaises(ValueError):
+            reader = RootFileReader({})
+
+        # Finally, try a good call
+        testfile = ROOT.TFile("testfile.root")
```
Comment (clelange) [inline tests/test_rootfilereader.py]: Can you use `path_to_file = make_tmp_root_file(close=True)` and delete the ROOT file at the end again? And also delete `test.txt`?

**Comment by AndreasAlbert** [comment]
I overhauled the output cleanup completely. It is now done using the `TestCase.addCleanup` method. This allows you to register a clean-up action that will be taken automatically, no matter if there is an exception or the test fails. At the end of each test method, `TestCase.doCleanups` is called to ensure that the files are not only cleaned up at the end of all tests, but also in between tests. Finally, I changed all temporary root files to use the dedicated function and made it so that the function automatically registers the temporary files for deletion.

**Pull request merged**